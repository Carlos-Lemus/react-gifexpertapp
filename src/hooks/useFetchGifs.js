import { useEffect, useState } from 'react';
import getGifs from '../helpers/getGifs';
import PropTypes from "prop-types";

export const useFecthGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {

        getGifs(category).then(gifs => [

            setState({
                data: gifs,
                loading: false
            })

        ]);

    }, [category]);

    return state;
}

useFecthGifs.propTypes = {
    category: PropTypes.string.isRequired
}