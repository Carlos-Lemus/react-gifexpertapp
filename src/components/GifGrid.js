import React, { useEffect, useState } from 'react';
import PropTypes from "prop-types";
import { useFecthGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from './GifGridItem';

const GifGrid = ( { category } ) => {

    const {data:gifs, loading} = useFecthGifs(category);
    

    return (
        <>
            <h3>{ category }</h3>
            {loading && <p className="animate__animated animate__flash animate__delay-2">Loading...</p>}

            { <ol className="card-grid">
                {
                    gifs.map( image => (
                        <GifGridItem key={image.id} {...image} />
                    ))
                }
            </ol> }

        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};

export default GifGrid;