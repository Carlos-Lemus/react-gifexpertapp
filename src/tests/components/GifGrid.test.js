import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFecthGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Probando el componente <GifGrid />", () => {

    const category = "Saitma"; 
    
    test('Haciendo snapshot sobre el componente <GifGrid />', () => {
        
        useFecthGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={category} /> );
        
        expect(wrapper).toMatchSnapshot();  
    })
    
    test('debe mostrar items cuando carga las imagenes', () => {
        
        const gifs = [{
            id: 1,
            url: "https://loqueseam",
            title: "gif test"
        }]
        
        useFecthGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category} /> );
        
        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find("p").exists() ).toBe(false);

    })
    

});