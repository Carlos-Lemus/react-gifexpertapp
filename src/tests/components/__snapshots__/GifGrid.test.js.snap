// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GifGrid /> Haciendo snapshot sobre el componente <GifGrid /> 1`] = `
<Fragment>
  <h3>
    Saitma
  </h3>
  <p
    className="animate__animated animate__flash animate__delay-2"
  >
    Loading...
  </p>
  <ol
    className="card-grid"
  />
</Fragment>
`;

exports[`Probando el componente <GifGrid /> debe mostrar items cuando carga las imagenes 1`] = `
<Fragment>
  <h3>
    Saitma
  </h3>
  <ol
    className="card-grid"
  >
    <GifGridItem
      id={1}
      key="1"
      title="gif test"
      url="https://loqueseam"
    />
  </ol>
</Fragment>
`;
